Mathlib/Data/Fintype/Units.lean

```diff
@@ -44,3 +44,10 @@ theorem Fintype.card_units [GroupWithZero α] [Fintype α] [Fintype αˣ] :
     have := Fintype.card_congr (Equiv.sumCompl (· = (0 : α))).symm
     rwa [Fintype.card_sum, add_comm, Fintype.card_subtype_eq] at this
 #align fintype.card_units Fintype.card_units
+
+theorem Fintype.card_units_add_one [GroupWithZero α] [Fintype α] [Fintype αˣ] :
+    Fintype.card αˣ + 1 = Fintype.card α := by
+  classical
+    rw [eq_comm, Fintype.card_congr (unitsEquivNeZero α)]
+    have := Fintype.card_congr (Equiv.sumCompl (· = (0 : α))).symm
+    rwa [Fintype.card_sum, add_comm, Fintype.card_subtype_eq] at this
```

Comment:
How about:
```suggestion
theorem Fintype.card_eq_card_units_add_one [GroupWithZero α] [Fintype α] :
    Fintype.card α = Fintype.card αˣ + 1 := by
  rw [Fintype.card_units, Nat.sub_add_cancel Fintype.card_pos]
```
Note that this drops the `[Fintype αˣ]` argument because this follows from `[Fintype α]` provided we also add the following:
```lean
noncomputable instance [Monoid α] [Fintype α] : Fintype αˣ := Fintype.ofFinite αˣ
```
In fact that should go above `Fintype.card_units` so that the `[Fintype αˣ]` argument there can also be dropped.

Reply:
@ocfnash Even better would have been to reprove `card_units` using `card_units_add_one`, that way we could use the cancellation lemma without the condition. Are we allowed to modify ported theorem proofs?

Reply:
(I just went ahead and did it, I assume I'll be told if it's a problem)